expressions.js

```diff
@@ -27,9 +27,37 @@ module.exports = {
   },
   // Encore => Vega redirects
   encoreBibPage: {
-      expr: /C__Rb(\d{8})__/,
+    expr: /C__Rb(\d{8})(__|~\$1|$)/,
     handler: (match) => `${VEGA_URL}/search/card?recordId=${match[1]}`
   },
+  languagesOtherThanEnglish: {
+    expr: /C__Sf:\((a|v)%20\|%20(u|y)\)(?:.*?)l:\(?([a-z]{3})\)?/,
+    handler: (match) => {
+      const materialTypes = match[1] + ',' + match[2]
+      const languageId = match[3]
+      return `${VEGA_URL}/search?query=*&searchType=everything&pageSize=10&languageIds=${languageId}&pageNum=0&materialTypeIds=${materialTypes}&sorting=publicationDate&sortOrder=desc`
+    }
+  },
+  authorOrTitleSearch: {
+    custom: (path) => {
+      console.log(path)
+      try {
+      const decodedPath = decodeURIComponent(path)
+      const regEx = /(?:[(]+([^-)]+)[^)]*)/g
+      const matches = [...decodedPath.matchAll(regEx)]
+      const searchTerms = encodeURI(matches
+        .map((match) => `"${match[1]}"`).join(' '))
+        return searchTerms
+      } catch (e) {
+        // this is here because one of the tests was causing a malformed URI error
```

Comment:
This is interesting — do you know where that happens? Is it from the `decodeURIComponent` call? If you could add the problematic URI as a reply here I'd like to take a look — probably I fat-thumbed the URI.

Reply:
Ooh I got it. l realized it was having an issue with the percent sign. Removed and now there is no error. I still think that the 404 is not sensitive enough though.

Reply:
/record=&%!^/ was the problematic path

Reply:
Good catch. Makes sense that's a bad URL